testdata/p4tc_samples_outputs/add_entry_1_example_control_blocks.c

```diff
@@ -161,6 +178,12 @@ if (/* hdr->ipv4.isValid() */
                 return TC_ACT_SHOT;
             }
         }
+
+        if (have_saved_proto) {
+            bpf_p4tc_skb_set_protocol(skb, &sa->set, saved_proto);
```

Comment:
Is setting the protocol and skb_meta required everytime or can we optimize this and set only when skb externs are called

Reply:
The setting of the protocol is a way to avoid `bpf_skb_adjust_room`'s limitation of only supporting IPv4 and IPv6
We want to avoid making the user aware of this because this would hurt usability and require them to know specifics about `bpf_skb_adjust_room` that shouldn't matter to them
So we chose to do this automatically without involving the user